### ISSUE TITLE
L7 port range tests always fail

### ISSUE THREAD

**Issue by dhess** [cilium/cilium#36774]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Version

equal or higher than v1.16.5 and lower than v1.17.0

### What happened?

When we run `cilium connectivity test`, the L7 port range checks always fail:

```
[=] [cilium-test-1] Test [client-egress-l7-method-port-range] [69/108]
...
2024-12-23T12:35:51.308129975Z time="2024-12-23T12:35:51Z" level=warning msg="Unable to add CiliumNetworkPolicy" ciliumNetworkPolicyName=client-egress-l7-http-method-port-range error="Invalid CiliumNetworkPolicy spec: L7 rules do not support port ranges" k8sApiVersion= k8sNamespace=cilium-test-1 subsys=policy-k8s-watcher
...
  ❌ Error finalizing 'client-egress-l7-method-port-range': timed out waiting for policy updates to be processed on Cilium agents: error with exec request (pod=kube-system/cilium-w8v5q, container=cilium-agent): command terminated with exit code 1: "Error: \n25 endpoints still not ready after 30.270633698s (0 failed)\n\n"
  🟥 [cilium-test-1] test client-egress-l7-method-port-range failed: setting up test: applying network policies: policies were not applied on all Cilium nodes in time: error with exec request (pod=kube-system/cilium-xlf2c, container=cilium-agent): command terminated with exit code 1: "Error: \n20 endpoints still not ready after 30.169530518s (0 failed)\n\n"

```

I feel like I must be missing something, because the docs clearly state that "Layer 7 rules do not currently support port ranges":

https://docs.cilium.io/en/v1.16/security/policy/language/#layer-7-examples

These failing tests take a long time to fail due to all the timeouts. So my questions are:

1. Under what circumstances are these tests expected to pass?
2. Do the failures have something to do with the fact that we're using Envoy and `l7Proxy: true`?
3. Can the connectivity test be changed to detect the scenario where the policy will not apply, and skip these tests in that scenario?

### How can we reproduce the issue?

`cilium connectivity test --test-namespace cilium-test --verbose --test "client-egress-l7-method-port-range"`

### Cilium Version

1.16.5

### Kernel Version

`Linux worker-d 6.6.64-talos #1 SMP Wed Dec 11 15:33:07 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux`

### Kubernetes Version

`v1.31.4`

### Sysdump

[cilium-sysdump-20241223-133706.zip](https://github.com/user-attachments/files/18230227/cilium-sysdump-20241223-133706.zip)

### Relevant log output

```shell

```

### Cilium Users Document

- [x] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by julianwiedmann** [comment]
:wave: hi! Are you potentially using a CLI version that doesn't include https://github.com/cilium/cilium/pull/36460 ?

**Comment by dhess** [comment]
I'm running `cilium-cli: 0.16.21 compiled with go1.23.3 on darwin/arm64`

**Comment by julianwiedmann** [comment]
Makes sense - this should be resolved with `0.16.22`. Please shout if it's still a problem when using that version!
